gear/form_qc_coordinator/src/python/form_qc_coordinator_app/main.py

```diff
@@ -0,0 +1,148 @@
+"""Defines Form QC Coordinator."""
+
+import logging
+from typing import Optional
+
+from flywheel import Client
+from flywheel.view_builder import ViewBuilder
+from flywheel_adaptor.subject_adaptor import (
+    ParticipantVisits,
+    SubjectAdaptor,
+)
+from flywheel_gear_toolkit import GearToolkitContext
+from gear_execution.gear_execution import (
+    ClientWrapper,
+    GearExecutionError,
+    InputFileWrapper,
+)
+from pandas import DataFrame
+
+from form_qc_coordinator_app.coordinator import QCCoordinator, QCGearInfo
+
+log = logging.getLogger(__name__)
+
+
+def update_file_tags(gear_context: GearToolkitContext,
+                     input_wrapper: InputFileWrapper):
+    """Add gear tag to file.
+
+    Args:
+        gear_context: Flywheel gear context
+        input_wrapper: gear input file wrapper
+    """
+
+    gear_name = gear_context.manifest.get('name', 'form-qc-coordinator')
+    gear_context.metadata.add_file_tags(input_wrapper.file_input,
+                                        tags=gear_name)
+
+
+def get_matching_visits(
+        *,
+        fw_client: Client,
+        container_id: str,
+        subject: str,
+        module: str,
+        date_col: str,
+        cutoff_date: Optional[str] = None) -> Optional[DataFrame]:
```

Comment:
Using the dataframe here forces some odd logic when running the QC checks b/c you have to iterate through all of the rows even after a condition occurs where checks shouldn't happen for subsequent visits. A queue ordered by visit date would be a better choice, because you can split the check code from the error code.

Reply:
Changed.